lib/manager/nuget/artifacts.ts

```diff
@@ -0,0 +1,129 @@
+import { id } from '../../datasource/nuget';
+import { logger } from '../../logger';
+import { ExecOptions, exec } from '../../util/exec';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../util/fs';
+import * as hostRules from '../../util/host-rules';
+import {
+  UpdateArtifact,
+  UpdateArtifactsConfig,
+  UpdateArtifactsResult,
+} from '../common';
+import { determineRegistries } from './util';
+
+async function authenticate(
+  packageFileName: string,
+  config: UpdateArtifactsConfig,
+  cmds: string[]
+): Promise<void> {
+  const registries = (
+    (await determineRegistries(packageFileName, config.localDir)) || []
+  ).filter((registry) => registry.name != null);
+  for (const registry of registries) {
+    const { username, password } = hostRules.find({
+      hostType: id,
+      url: registry.url,
+    });
+    if (username && password) {
+      // Add registry credentials from host rules.
+      cmds.unshift(
+        `dotnet nuget update source ${registry.name} --username ${username} --password ${password} --store-password-in-clear-text`
+      );
+      // Ensure that credentials are removed as soon as not necessary anymore.
+      cmds.push(
+        `dotnet nuget update source ${registry.name} --username '' --password '' --store-password-in-clear-text`
+      );
+    }
+  }
+}
+
+async function runDotnetRestore(
+  packageFileName: string,
+  config: UpdateArtifactsConfig
+): Promise<void> {
+  const execOptions: ExecOptions = {
+    cwdFile: packageFileName,
+    docker: {
+      image: 'renovate/dotnet',
+    },
+  };
+  const cmds = [`dotnet restore --force-evaluate ${packageFileName}`];
+  await authenticate(packageFileName, config, cmds);
+  logger.debug({ cmd: cmds }, 'dotnet command');
+  await exec(cmds, execOptions);
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  newPackageFileContent,
+  config,
+  updatedDeps,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`nuget.updateArtifacts(${packageFileName})`);
+
+  if (!/[cs|vb|fs]proj$/i.test(packageFileName)) {
```

Comment:
```suggestion
  if (!/(?:cs|vb|fs)proj$/i.test(packageFileName)) {
```

Same as the filematch regex

https://github.com/renovatebot/renovate/blob/f52fe033de5fae6be0155c3b36fad1ec6f722f01/lib/manager/nuget/index.ts#L8-L12